--- a/ogg123/status.c
+++ b/ogg123/status.c
@@ -24,7 +24,9 @@
 #include <pthread.h>
 
 #ifdef HAVE_UNISTD_H
+#ifndef _WIN32
 #include <sys/ioctl.h>
+#endif
 #include <unistd.h>
 #endif
 
@@ -168,7 +170,7 @@
     stats++;
   }
   
-#ifdef HAVE_UNISTD_H
+#if defined(HAVE_UNISTD_H) && !defined(_WIN32)
   /* If the line would break in the console, truncate it to avoid the break,
      and indicate the truncation by adding points of ellipsis */
   struct winsize max;
